crates/bevy_ecs/src/bundle.rs

```diff
@@ -9,17 +12,36 @@ use crate::{
 use bevy_ecs_macros::all_tuples;
 use std::{any::TypeId, collections::HashMap};
 
-/// An ordered collection of components, commonly used for spawning entities, and adding and
-/// removing components in bulk.
+/// An ordered collection of components.
 ///
-/// In order to query for components in a bundle use [crate::query::WithBundle].
+/// Commonly used for spawning entities and adding and removing components in bulk. This
+/// trait is automatically implemented for tuples of components: `(ComponentA, ComponentB)`
+/// is a very convenient shorthand when working with one-off collections of components. Note
+/// that both `()` and `(ComponentA, )` are valid tuples.
 ///
-/// Typically, you will simply use `#[derive(Bundle)]` when creating your own `Bundle`.
-/// The `Bundle` trait is automatically implemented for tuples of components:
-/// `(ComponentA, ComponentB)` is a very convenient shorthand when working with one-off collections
-/// of components. Note that both `()` and `(ComponentA, )` are valid tuples.
+/// In order to query for components in a bundle use
+/// [`query::WithBundle`](crate::query::WithBundle).
```

Comment:
This deserves a more verbose explanation: that's a query *filter* only.

I would consider "querying for components" to refer to fetching their data by default, and new users will be surprised when they cannot put this in the first type argument position of `Query`.

Reply:
This was part of the old docs that I didn't bother to edit. I changed it to mention the use-case of the filter, and a link to `WithBundle`, where I also added a doc summary.